templates/common/baremetal/files/NetworkManager-static-dhcp.yaml

```diff
@@ -64,5 +64,6 @@ contents:
 
     if [ -n "${DHCP4_HOST_NAME:-}" ]
     then
-        hostnamectl set-hostname --static --transient "$DHCP4_HOST_NAME"
+        systemd-run --property=Type=oneshot --unit static-dhcp-hostnamectl -Pq \
```

Comment:
might be totally wrong here but systemd-run docs say ` If the --unit= option is specified, the COMMAND may be omitted. In this case, systemd-run creates only a .path, .socket, or .timer unit that triggers the specified unit` which makes me feel like `hostnamectl...` is being omitted, no? The hypershift failures look strange enough to me that this might be an actual issue.

Reply:
I'm using the same invocation as https://github.com/openshift/machine-config-operator/blob/16969b5fb00eea7cb96f5a0aa88b80c559e03653/templates/common/on-prem/files/resolv-prepender.yaml#L33 where it is doing what we need.

I _think_ what the docs are saying is that if you specify an existing unit then you don't have to provide a command and it will just run the unit, but in this case the unit doesn't exist so it runs the command using the unit name we provide.